File: Kotlin/java-example/src/main/java/org/delivery/ex01/Exam01.java
package org.delivery.ex01;

public class Exam01 {

    //생성자 메서드
    public Exam01(){

        /** [코드작성]
         * 기본적으로 여기가 static이기 때문에 변수를 선언한다고 하면,
         * public static or static으로 다 선언해줘야 한다.
         * 그래서 아래에 main만 하위에서 호출을 하도록 할 것이다.
         */

        //===================== 변수선언 =====================
        //자바에는 primitive 타입과 reference 타입이 있다

        String name = "김예나";
        String format = "사용자의 이름은 : %s";
        String result = String.format(format, name);

        int age = 10;       //primitive 타입; 기본값은 0이다.
        Integer _age = 20;  //reference 타입; 기본값은 null이다.

        double d = 10d;
        Double _d = 20.0;

        float f = 20f;
        Float _f = 20f;

        long l = 10L;
        Long _l = 10L;

        boolean bool = true;
        Boolean _bool = true;

        //===================== 출력 =====================
        System.out.println(name);
        System.out.println("사용자의 이름은: " + name);
        System.out.println(result);

        System.out.println(age);
        System.out.println(_age);
        System.out.println(d);
        System.out.println(_d);
        System.out.println(f);
        System.out.println(_f);
        System.out.println(l);
        System.out.println(_l);
        System.out.println(bool);
        System.out.println(_bool);
    }

    public static void main(String[] args) {

        new Exam01();
    }
}

File: Kotlin/kotlin-example/src/main/kotlin/ex01/Exam01.kt
package org.delivery.ex01

fun main() {

    //===================== 변수선언 =====================
    //Kotlin에서는 변수선언할 때 2가지 방식이 가능하다
        //1. var : 변할 수 있는 객체(가변객체; mutable)
        //2. val : 값을 할당하면 변경이 불가능(불변 객체; immutable; 자바의 final과 같음)
    // : [타입]       << 선언 방식
    //Kotlin은 모든 게 reference 타입이다. primitive 타입이 없다 << 모든 것은 객체로 관리된다


    val name: String = "김에나"
    var _name: String = "김예나"
    val n = "김예나" //타입 추론이 된다

    val result = "나의 이름은 : $name"

    var num1 = 10
    num1 = 20; // var은 가변이기에 값 변경O

    var _num1: Int = 10
    _num1 = 20;

    val num2 = 30
//    num2 = 30; val은 불변이기에 값 변경X

    var d: Double = 20.0

    var f: Float = 10f

    var b: Boolean = true

    //===================== 출력 =====================
    println(name)
    println(_name)
    println(n)
    println("사용자의 이름은: $name")
    println("사용자의 이름은: ${if(true)name else null}")
    println(result)

    println(num1)
    println(_num1)
    println(num2)

    println(d)
    println(f)
    println(b)
}